Update to the Patreon v2 API

martincostello opened the pull request:
Updates the Patreon provider to use the v2 API.

Resolves #433.

The only thing I'm not sure about is whether this would be part of 3.1.2 as a "fix", or would be a follow-up 3.2.0 as it adds new functionality.

martincostello committed:
Update to the Patreon v2 API

Updates the Patreon provider to use the v2 API.
Resolves #433.

martincostello commented:
@kevinchalet What's your opinion on whether this should be part of 3.1.2, or should wait until a 3.2.0?

kevinchalet commented on src/AspNet.Security.OAuth.Patreon/PatreonAuthenticationHandler.cs:
```diff
@@ -34,7 +35,19 @@ public class PatreonAuthenticationHandler : OAuthHandler<PatreonAuthenticationOp
-            using var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            string endpoint = Options.UserInformationEndpoint;
+
+            if (Options.Fields?.Count > 0)
```
Looks like these two collections have no setter and are always non-null, so the `?` is probably not necessary.

martincostello replied:
Good spot! Will amend.

kevinchalet reviewed (commented):
> @kevinchalet What's your opinion on whether this should be part of 3.1.2, or should wait until a 3.2.0?

The v1 API is still working, right? If you plan to release the Auth0 provider as part of 3.2.0, then maybe it can wait until then?

martincostello commented:
> The v1 API is still working, right?

It's still there, but it'll go away. I'm just not sure when.

>  If you plan to release the Auth0 provider

Auth0? Or did you mean the Okta one I added earlier? If so, I was planning on making that part of 3.1.2 as we've added new providers in patch releases before.

kevinchalet commented:
> Auth0? Or did you mean the Okta one I added earlier?

lol, yeah, Okta (I should really try to sleep... 🤣)

> If so, I was planning on making that part of 3.1.2 as we've added new providers in patch releases before.

Oh, then feel free to ship this change as part of 3.1.2.

martincostello committed:
Remove null checks

Remove redundant null checks.
